lib/Translation/ExportVerilog/ExportVerilogFIRRTL.cpp

```diff
@@ -1799,6 +1799,13 @@ void ModuleEmitter::collectNamesEmitDecls(Block &block) {
                           .str();
           addName(result, name);
         }
+      } else if (auto memory = dyn_cast<MemOp>(&op)) {
+        if (nameAttr) {
+          auto name = (nameAttr.getValue() + "_" +
+                       memory.getPortNameStr(result.getResultNumber()))
+                          .str();
```

Comment:
Ignore my comment below, I think I see.

Reply:
This is using Lattner's approach for instances with multiple results. The actual result is `memory#0` or whatever, but it prints using the portname information as `memory_ + portNames(0)` or whatever.